tasks/species_typing/task_tbprofiler_output_parsing.wdl

```diff
@@ -0,0 +1,484 @@
+                          "rpoB": ["rifampicin"], "rpoC": ["rifampicin"], "rpsA": ["pyrazinamide"],
+                          "rpsL": ["streptomycin"], "rrl": ["linezolid"], "rrs": ["streptomycin", "amikacin",
+                          "aminoglycosides", "capreomycin", "kanamycin"], "Rv0678": ["bedaquiline",
+                          "clofazimine"], "thyA": ["para-aminosalicylic_acid"], "thyX": ["para-aminosalicylic_acid"],
+                          "tlyA": ["capreomycin"]
+                         }
+
+    # lookup dictionary - gene to locus tag (https://github.com/jodyphelan/TBProfiler/blob/master/db/tbdb.bed)
+    gene_to_locus_tag = {"ahpC":"Rv2428", "ald":"Rv2780", "alr": "Rv3423c",
+                          "ddn": "Rv3547", "eis": "Rv2416c", "embA": "Rv3794",
+                          "embB": "Rv3795", "embC": "Rv3793", "embR": "Rv1267c",
+                          "ethA": "Rv3854c", "ethR": "Rv3855", "fabG1": "Rv1483",
+                          "fbiA": "Rv3261", "fgd1": "Rv0407", "folC": "Rv2447c",
+                          "gid": "Rv3919c", "gyrA": "Rv0006", "gyrB": "Rv0005",
+                          "inhA": "Rv1484","kasA": "Rv2245", "katG": "Rv1908c", 
+                          "panD": "Rv3601c", "pncA": "Rv2043c", "ribD": "Rv2671",
+                          "rplC": "Rv0701", "rpoB": "Rv0667", "rpoC": "Rv0668", 
+                          "rpsA": "Rv1630", "rpsL": "Rv0682", "rrl": "EBG00000313339", 
+                          "rrs": "EBG00000313325", "Rv0678": "Rv0678", "thyA": "Rv2764c",
+                          "thyX": "Rv2754c", "tlyA": "Rv1694"
+                         }
+
+    ## Auxiliary Functions ##
+
+    def get_codon(protein_mut):
+      """
+      This function recieves a protein mutation (e.g. 'p.Met291Ile')
+      and returns the codon (numerical part) where that mutation occurs
+      in Int format.
+      """
+      pattern = r"\.\D+(\d+)"
+
+      match = re.search(pattern, protein_mut)
+      if match:
+        position = match.group(1)
+        return int(position)
+      return 0
+
+    def decipher_looker(annotation):
+      """
+      This function takes the annotation of resistance by TBProfiler and 
+      returns simple R (resistant), U (uncertain) and S (susceptible) notations
+      """
+      if annotation == "Not assoc w R":
+        return "S"
+      elif annotation == "Uncertain significance":
+        return "U"
+      elif annotation == "Assoc w R - interim":
+        return "R-interim"
+      elif annotation == "Assoc w R":
+        return "R"
+      else:
+        return "S"
+    
+    def decipher_MDL(annotation):
+      """
+      This function takes the annotation of resistance by TBProfiler and 
+      returns simple R (resistant), U (uncertain) and S (susceptible) notations
+      """
+      if annotation == "Not assoc w R":
+        return "S"
+      elif annotation == "Uncertain significance":
+        return "S"
```

Comment:
I guess I don't understand enough about this to know why the same annotations are returning different things based on the function?